Sources/Plasma/CoreLib/plFileSystem.cpp

```diff
@@ -460,7 +469,36 @@ plFileName plFileSystem::GetUserDataPath()
 
         _userData = plFileName::Join(ST::string::from_wchar(path), plProduct::LongName());
 #else
-        _userData = plFileName::Join(getenv("HOME"), "." + plProduct::LongName());
+        ST::string dir;
+#  if HS_BUILD_FOR_APPLE
+        char home[256] = "";
+        if (__builtin_available(macOS 10.12, *)) {
+            sysdir_search_path_enumeration_state state;
+            state = sysdir_start_search_path_enumeration(SYSDIR_DIRECTORY_APPLICATION_SUPPORT, SYSDIR_DOMAIN_MASK_USER);
+            state = sysdir_get_next_search_path_enumeration(state, home);
+        } else {
+            #pragma clang diagnostic push
+            #pragma clang diagnostic ignored "-Wdeprecated-declarations"
+            NSSearchPathEnumerationState state;
+            state = NSStartSearchPathEnumeration(NSApplicationSupportDirectory, NSUserDomainMask);
+            state = NSGetNextSearchPathEnumeration(state, home);
+            #pragma clang diagnostic pop
+        }
+        dir = home;
+#  else
+        dir = "~" PATH_SEPARATOR_STR ".config";
+#  endif
+
+        wordexp_t p;
+        wordexp(dir.c_str(), &p, 0);
+        dir = p.we_wordv[0];
+        for (int i = 1; i < p.we_wordc; i++) {
+            dir += " ";
+            dir += p.we_wordv[i];
```

Comment:
This is non-optimal in terms of performance, FWIW.

Reply:
This case only happens when there are spaces in the path. Which is probably rare on Linux, but macOS uses a folder called "Application Support" 🙄